File: walletcore/src/main/java/com/mycelium/wapi/wallet/GenericAddress.java
package com.mycelium.wapi.wallet;

import com.mrd.bitlib.model.AddressType;
import com.mrd.bitlib.model.NetworkParameters;
import com.mrd.bitlib.model.hdpath.HdKeyPath;
import com.mycelium.wapi.wallet.coins.CryptoCurrency;

import java.io.Serializable;

public interface GenericAddress extends Serializable{
    CryptoCurrency getCoinType();
    AddressType getType();
    long getId();
    HdKeyPath getBip32Path();
}

File: walletcore/src/main/java/com/mycelium/wapi/wallet/GenericAddress.kt
package com.mycelium.wapi.wallet

import com.mycelium.wapi.wallet.coins.CryptoCurrency

import java.io.Serializable

interface GenericAddress : Serializable {
    val coinType: CryptoCurrency
    val id: Long

    // An address for the particular asset could have some subtypes.
    // For example, for BTC we have
    // To have an ability to detect and compare types in generic way
    // the subType is stored as string
    fun getSubType(): String

    fun getBytes():ByteArray
}
